godot/scripts/entities/Player.gd

```diff
@@ -18,6 +20,9 @@ enum State {
 
 var _current_interactable = null
 var _current_state = State.IDLE
+var _is_facing_right = true
+
+var _is_ready_to_spawn = true
```

Comment:
You may want _is_ready_to_spawn to be a dictionary<tool, bool>, so that you can track the ready state of the spawners independently. For example, if the player had cleaver and flamethrower, they could alternate the buttons to cut down the cooldown time.